### ISSUE TITLE
VSC plugin Gauge version 0.0.21 - debugger does not work

### ISSUE THREAD

**Issue by sebor255** [getgauge/gauge-vscode#726]
### Description
The newest version of VSC's plugin Gauge does not work when debugging. 
When a user wants to start debugging and presses "Debug" over the scenario the error appears: 

"Failed to start debugger: The debugger does not work for a stand alone file. Please open the folder C:\MyNewProject"
Tested with various configurations of getgauge python package and gauge versions. The behaviour is the same. 

Works fine with VSC's Gauge plugin 0.0.20.  

### Action
Run debugger from the specification.

### Versions
Python 3.6.8
Gauge version: 1.4.1
Commit Hash: a82478a

Plugins
-------
html-report (4.1.2)
json-report (0.3.6)
python (0.3.10)
screenshot (0.1.0)
spectacle (0.1.4)
xml-report (0.2.3)

#### Gauge output 

Python: 3.6.8
Runner Ready for Debugging
Error: Tests failed.

#### Operating System information
Windows 10 x64

#### IDE information
VSC 1.60.0

**Comment by zabil** [comment]
The debugger works on the latest version on a Mac. I see that this an error on a windows machine.

<img width="981" alt="Screenshot 2021-09-09 at 13 38 45" src="https://user-images.githubusercontent.com/54427/132690933-3ee9a25e-7009-4085-9bc7-dd47f06ad8c2.png">

Can you check (and share) the Gauge log folders or Visual studio Code's "Output" tab for more information?

**Comment by vjacobjo** [comment]
Hi,
I have a similar issue, which I had reported on Stack overflow here: https://stackoverflow.com/questions/69104671/unable-to-start-the-vscode-debugger-with-the-gauge-framework

I've also enabled the debug files and I've got the outputs here: 
[lsp.log](https://github.com/getgauge/gauge-vscode/files/7137956/lsp.log)
[gauge.log](https://github.com/getgauge/gauge-vscode/files/7137957/gauge.log)

My machine is also a Windows 10 Professional OS  x64.

**Comment by vjacobjo** [comment]
I was able to revert to version 0.0.20. And it worked there.

![image](https://user-images.githubusercontent.com/7102169/132759481-7eeda6c6-caeb-4bd3-9be8-171be47768ba.png)

Here are the log files that I've generated:
[lsp_0.0.20.log](https://github.com/getgauge/gauge-vscode/files/7139641/lsp_0.0.20.log)
[gauge_0.0.20.log](https://github.com/getgauge/gauge-vscode/files/7139642/gauge_0.0.20.log)

**Comment by zabil** [comment]
To rule out any version issues with python or the python vs code plugin can you try using python version 3.9.7 and python vscode plugin version v2021.9.1191016588?

**Comment by vjacobjo** [comment]
Test 1: 
- Python version 3.9.7
- Python for VSCode extension (ms-python.python) v2021.9.1191016588.
- Gauge plugin for VSCode v0.0.21

I was ABLE to reproduce the error.

[lsp_test1.log](https://github.com/getgauge/gauge-vscode/files/7143970/lsp_test1.log)
[gauge_test1.log](https://github.com/getgauge/gauge-vscode/files/7143971/gauge_test1.log)


Test 2:
- Python version 3.9.7
- Python for VSCode extension (ms-python.python) v2021.9.1191016588.
- Gauge plugin for VSCode v0.0.20

I was UNABLE to reproduce the error: Debugger was able to work successfully.
![image](https://user-images.githubusercontent.com/7102169/132858391-c6440e01-cf70-4346-960a-01dd4e22fd76.png)

Log files are here
[lsp_test2.log](https://github.com/getgauge/gauge-vscode/files/7143987/lsp_test2.log)
[gauge_test2.log](https://github.com/getgauge/gauge-vscode/files/7143988/gauge_test2.log)

**Comment by zbrea** [comment]
I use gauge-java. I have the same problem with this version
![debug](https://user-images.githubusercontent.com/59898499/133001105-f3d5e90e-13d3-40d8-9749-f804be820e4f.JPG)

**Comment by sriv** [comment]
https://github.com/getgauge/gauge-vscode/pull/700 is the change that was included in gauge-vscode 0.0.21, and it intended to add 'Debug' and `Run' options in the explorer tree. I will investigate if this could be a cause, but from my testing so far this hasn't been the case.

**Comment by at6ue** [comment]
It appears to be due to [this change](https://github.com/getgauge/gauge-vscode/pull/700/files#diff-9c5c7fb24e3193de93931bd71597a98a1f7e241e33b9fbbec072d91832d146e2L138)... sorry for inconvenience.
This error occurs when the spec file to be executed is not contained in the workspace folder, but as far as I see @vjacobjo's log, the spec is executed from the same folder.

It may have lost its compatibility with windows file paths in some conditions. Though I'm not sure if it relates to this case, I found the line below from the lsp log;
```
[Gauge] [DEBUG] jsonrpc2: --> request #4: textDocument/codeLens: {"textDocument":{"uri":"file:///c%!!(MISSING)A(MISSING)/Projects/GaugeFramework/ias_net_automation/specs/testData.spec"}}
```

**Comment by zabil** [comment]
This should now be fixed in https://github.com/getgauge/gauge-vscode/releases/tag/v0.0.22

Thanks @at6ue!

**Comment by vjacobjo** [comment]
I can confirm that this issue has been fixed in v0.0.22. Thank you so much everyone :)
